### PR TITLE
PythonInline geometry element

### PR DESCRIPTION
**Pull request by mdzik**
I added the PythonInline geometry element. Bit experimental*, but with this making "fancy" geometry is much cleaner: no need to outside script.
Performance should be comparable to the rest of the code, function is parsed/compiled once and then executed.
Additional you have state-full  Python environment in whole "< nx= ny=>" box

*) might be unstable, I hope I fixed all thread/gis related issues, but one never knows. Should have no impact on solving part of the solver. Python is finalized after processing geometry block.

### CODE REVIEW

File: src/Solver.cpp.Rt [hunk 1]
```diff
@@ -543,13 +543,61 @@ int Solver::getComponentIntoBuffer(const char* comp, real_t* &buf, long int*  di
...
+
+    
+    offsets[0] = region.dx;
+    offsets[1] = region.dy;
+    offsets[2] = region.dz;
+
+	output("Providing quantitie %s to buffer..\n", comp);
+    bool somethingWritten = false;
+<?R 
+    for (q in rows(Quantities)) {
```
Comment (llaniewski) [inline src/Solver.cpp.Rt]: you need some ifdef Adjoint here. like in other places with Quantities.

**Comment by mdzik** [comment]
ifdef should "protect" against calling Solver->Get_q$name ? But preserve memory allocation etc.?

static_cast: http://stackoverflow.com/a/22330938

**Comment by llaniewski** [comment]
ifdef should switch off parts of code that are adjoint specific. There is a helper function ifdef in R, which takes a bool and injects the `#ifdef ADJOINT`, you can see the usage here:

https://github.com/CFD-GO/TCLB/blob/a5885da3c6d67d2851da5ffa0a94ef1729d31873/src/Catalyst.cpp.Rt#L126

**Comment by mdzik** [comment]
```c
      <?R for (q in rows(Quantities)) { ifdef(q$adjoint); ?>
/////////////////////////////////////////////
      <?R
	};
	ifdef();
?>
```
why closing ifdef(); is after R for loop?

**Comment by llaniewski** [comment]
because it have to close the eventual opened `#ifdef`'s

**Comment by mdzik** [comment]
I'm not that dumb ;) 
But why AFTER closing _for_ {}?
I would expect  
```R
for (q in rows(Quantities)) {
       ifdef(q$adjoint); 
#sth
       ifdef();
};
?>
```

**Comment by mdzik** [comment]
ops, last one was supposed to go to separate branch.

**Pull request merged**